### ISSUE TITLE
BUG: Saving Brick-By-Brick models kicks users to the login screen

### ISSUE THREAD

**Issue by MustacheMan2** [DarkflameUniverse/DarkflameServer#939]
### Make sure you've done the following:

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### DarkflameServer Version

1789ec7

### Issue Description

This has happened for quite some time now (for months, but it was working fine at one point), but only today have I found out that it's not just me. After placing some bricks in brick building mode on a property and pressing save, it shows the loading bar for less than 2 seconds before booting the user to the login screen. Requests for the models were shown on the HTTP server hosted locally but the instance crashes before the server is able to save the models, so they don't ever load in on a relog. Tried it on several characters with no success, although existing models do load as usual. I wanted to run the command to fix broken models again but I couldn't find the exact command anywhere on GitHub.

Here are the logs for the most recent crash.
[WorldServer_1151_4_1672805355.log](https://github.com/DarkflameUniverse/DarkflameServer/files/10341384/WorldServer_1151_4_1672805355.log)
[crash_World_314332.log](https://github.com/DarkflameUniverse/DarkflameServer/files/10341386/crash_World_314332.log)

### Reproduction Steps

On a property, place any amount of bricks in brick building mode and press the place models arrow in the corner of the screen. (With the HTTP python script running, and patch/ugc server ip set to localhost)

### Expected Behavior

Brick models should save to the server without crashing, as they once did.

### Environment

ubuntu server

**Comment by EmosewaMC** [comment]
Unable to reproduce on main.  Brick models are able to be placed without error.

**Comment by EmosewaMC** [comment]
Some steps to resolve the issue: increase the `last_object_id` in your `object_id_tracker` by 1000 and see if that resolves the issue

**Comment by EmosewaMC** [comment]
My mistake, should have specified where this is.  This is a table in your mysql database

**Comment by MustacheMan2** [comment]
Increased the value by 1000 and the issue was not fixed.

**Comment by EmosewaMC** [comment]
Does this happen with standard models as well?

**Comment by MustacheMan2** [comment]
standard models can be saved without issue.

**Comment by EmosewaMC** [comment]
hmm, lets try this.  Im going to create a branch that will try and catch this error, log it to the console, and then crash the server.  I'll reply again with the branch name.  Switch to that branch, reproduce the issue and post the logs.  This will help narrow down what happened to get the server to a weird state that other servers do not experience

**Comment by EmosewaMC** [comment]
https://github.com/DarkflameUniverse/DarkflameServer/tree/MM12-Testing-branch

Switch to this branch and reproduce the error - this should catch the issue

**Comment by MustacheMan2** [comment]
couldn't get it to build. Here are some errors (it kept going on so I interrupted it).

```
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [_deps/googletest-build/googletest/CMakeFiles/gtest.dir/build.make:76: _deps/googletest-build/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3568: _deps/googletest-build/googletest/CMakeFiles/gtest.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dPhysics/CMakeFiles/dPhysics.dir/build.make:160: dPhysics/CMakeFiles/dPhysics.dir/dpWorld.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dPhysics/CMakeFiles/dPhysics.dir/build.make:132: dPhysics/CMakeFiles/dPhysics.dir/dpShapeBox.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dPhysics/CMakeFiles/dPhysics.dir/build.make:118: dPhysics/CMakeFiles/dPhysics.dir/dpShapeBase.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dNavigation/CMakeFiles/dNavigation.dir/build.make:76: dNavigation/CMakeFiles/dNavigation.dir/dNavMesh.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dZoneManager/CMakeFiles/dZoneManager.dir/build.make:77: dZoneManager/CMakeFiles/dZoneManager.dir/cmake_pch.hxx.gch] Error 1
make[2]: *** Deleting file 'dZoneManager/CMakeFiles/dZoneManager.dir/cmake_pch.hxx.gch'
make[1]: *** [CMakeFiles/Makefile2:3192: dZoneManager/CMakeFiles/dZoneManager.dir/all] Error 2
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dPhysics/CMakeFiles/dPhysics.dir/build.make:76: dPhysics/CMakeFiles/dPhysics.dir/dpCollisionChecks.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dPhysics/CMakeFiles/dPhysics.dir/build.make:90: dPhysics/CMakeFiles/dPhysics.dir/dpEntity.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dNavigation/CMakeFiles/dNavigation.dir/build.make:90: dNavigation/CMakeFiles/dNavigation.dir/dTerrain/RawFile.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[5]: *** [CMakeFiles/mariadbcpp.dir/build.make:90: CMakeFiles/mariadbcpp.dir/src/DriverManager.cpp.o] Error 1
make[5]: *** Waiting for unfinished jobs....
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[5]: *** [CMakeFiles/mariadbcpp.dir/build.make:538: CMakeFiles/mariadbcpp.dir/src/pool/Pools.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dPhysics/CMakeFiles/dPhysics.dir/build.make:104: dPhysics/CMakeFiles/dPhysics.dir/dpGrid.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[5]: *** [CMakeFiles/mariadbcpp.dir/build.make:244: CMakeFiles/mariadbcpp.dir/src/MariaDbSavepoint.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[5]: *** [CMakeFiles/mariadbcpp.dir/build.make:174: CMakeFiles/mariadbcpp.dir/src/MariaDbConnection.cpp.o] Error 1
```

**Comment by EmosewaMC** [comment]
Can I get the errors?

**Comment by EmosewaMC** [comment]
ahh I believe i know what happened.  If you built with build.sh it likely crashed on your system because I can build with -j.  Removed that from build.sh so it builds 1 job at a time

**Comment by MustacheMan2** [comment]
alright it's been compiling for hours now and its gone from frequent errors to all errors and no progress.

**Comment by EmosewaMC** [comment]
thats, odd because I just compiled this branch just fine on my local machine.  not sure what else to do.  
![image](https://user-images.githubusercontent.com/39972741/211178674-2a41eb7c-ea2d-4f33-83b5-578664407dcb.png)

**Comment by EmosewaMC** [comment]
What specifically does this get stuck on?

**Comment by MustacheMan2** [comment]
> What specifically does this get stuck on? The errors above are not errors helpful to me as they are just "I cancelled the build" errors. Im looking for red text during compilation. If the compilation is just taking a long time, check that you aren't out or memory / resources to make the compile fail.

It just goes on and on, but there's plenty of red text. Started it again with less background processes but it could take a very long time for it to finally stop. Here is some of what was being posted.

```
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:461: dGame/CMakeFiles/dGame.dir/dBehaviors/ChangeIdleFlagsBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:605: dGame/CMakeFiles/dGame.dir/dBehaviors/EndBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dCommon/CMakeFiles/dCommon.dir/build.make:333: dCommon/CMakeFiles/dCommon.dir/ZCompression.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:509: dGame/CMakeFiles/dGame.dir/dBehaviors/ClearTargetBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:301: dGame/CMakeFiles/dGame.dir/dBehaviors/AttackDelayBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:237: dGame/CMakeFiles/dGame.dir/dBehaviors/AirMovementBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:493: dGame/CMakeFiles/dGame.dir/dBehaviors/ChargeUpBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:173: dGame/CMakeFiles/dGame.dir/TeamManager.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dCommon/CMakeFiles/dCommon.dir/build.make:429: dCommon/CMakeFiles/dCommon.dir/dClient/AssetManager.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:557: dGame/CMakeFiles/dGame.dir/dBehaviors/DarkInspirationBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:253: dGame/CMakeFiles/dGame.dir/dBehaviors/AndBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dCommon/CMakeFiles/dCommon.dir/build.make:397: dCommon/CMakeFiles/dCommon.dir/dClient/PackIndex.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:413: dGame/CMakeFiles/dGame.dir/dBehaviors/BuffBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:381: dGame/CMakeFiles/dGame.dir/dBehaviors/BehaviorTemplates.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dCommon/CMakeFiles/dCommon.dir/build.make:269: dCommon/CMakeFiles/dCommon.dir/NiPoint3.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:397: dGame/CMakeFiles/dGame.dir/dBehaviors/BlockBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:685: dGame/CMakeFiles/dGame.dir/dBehaviors/InterruptBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:541: dGame/CMakeFiles/dGame.dir/dBehaviors/DamageReductionBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:621: dGame/CMakeFiles/dGame.dir/dBehaviors/ForceMovementBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:525: dGame/CMakeFiles/dGame.dir/dBehaviors/DamageAbsorptionBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
make[2]: *** [dGame/CMakeFiles/dGame.dir/build.make:701: dGame/CMakeFiles/dGame.dir/dBehaviors/JetPackBehavior.cpp.o] Error 1
c++: fatal error: Killed signal terminated program cc1plus
compilation terminated.
```

**Comment by EmosewaMC** [comment]
How are you building the server?

**Comment by MustacheMan2** [comment]
just with build.sh as usual

**Comment by EmosewaMC** [comment]
can you try navigating to build and running `make clean`, `rm CMakeCache.txt`, `cmake ..` and finally `make -j2`?  Ive run quite a few builds on an empty directory and have not had any issues so I am unsure whats going on, but this may get you to compile

**Comment by EmosewaMC** [comment]
the next time you see these errors, run `dmesg` immediately after the failure and see if a helpful error message shows up

**Comment by MustacheMan2** [comment]
ok so I was able to build it after doing that and without errors, but now the launchpads are broken (or at least the property ones) as the player just holds up the rocket without the property ui, so I can't visit the properties in the first place.

**Comment by EmosewaMC** [comment]
Im also experiencing that.  Investigating

**Comment by EmosewaMC** [comment]
Ive merged the fix into that branch.  Go ahead and pull and rebuild

**Comment by MustacheMan2** [comment]
[WorldServer_1350_3_1673151070.log](https://github.com/DarkflameUniverse/DarkflameServer/files/10367712/WorldServer_1350_3_1673151070.log)
[crash_World_314332.log](https://github.com/DarkflameUniverse/DarkflameServer/files/10367713/crash_World_314332.log)

**Comment by EmosewaMC** [comment]
Pushed some log changes.  This didnt catch the error so i need to pin down what query is erroring

**Comment by MustacheMan2** [comment]
Updated, and funny enough I was able to save brick models successfully on two properties. Would assume this is still problematic on main though and it was still crashing up to now.

**Comment by EmosewaMC** [comment]
thats expected

**Comment by MustacheMan2** [comment]
[WorldServer_1450_6_1673152553.log](https://github.com/DarkflameUniverse/DarkflameServer/files/10367732/WorldServer_1450_6_1673152553.log)

**Comment by EmosewaMC** [comment]
ah ha!  found it.

**Comment by EmosewaMC** [comment]
can you go into your mysql instance and run the following command in the darkflame database
`describe properties_contents`

**Comment by MustacheMan2** [comment]
![image](https://user-images.githubusercontent.com/48499798/211180735-b1225caa-4b35-4268-8352-cb2058aa583b.png)

**Comment by EmosewaMC** [comment]
yes there are missing columns, hence the error so lets get the migrations folder fixed.  Firstly verify the the folder `build/migrations/dlu` has 8 files in it, each pre-pended with a number 0-7.  There should be exactly 8 there, but if there arent and they arent being moved through cmake, run this command in build
`cp ../migrations/dlu/* migrations/dlu/`
Then run the following in your database to re-run all migrations `TRUNCATE TABLE migration_history;`.  Migrations are idempotent (aside from column additions which will error, but can be ignored) so no data should be erroneously altered.  Run master server again and then no error should happen on your main branch

**Comment by EmosewaMC** [comment]
Ah I remember what caused this and it has been resolved with #911 .  can be fixed by just re-running migration 6 which adds all these columns.  Once that migration is run the issue will be resolved

**Comment by MustacheMan2** [comment]
what's the name of the migration?

**Comment by MustacheMan2** [comment]
alright ran the migration, but the launchpad bug is also present on the latest commit on main, so I can't test the properties

**Comment by EmosewaMC** [comment]
> alright ran the migration, but the launchpad bug is also present on the latest commit on main, so I can't test the properties

Waiting on someone to review #946

**Comment by EmosewaMC** [comment]
if `describe properties_contents;` now has 18 columns you should be fine now.

**Comment by EmosewaMC** [comment]
If you want to just teleport to your property, you can /gmadditem 10292 and use said item to get transported to your Gnarled Forest property while the ui issues are reviewed

**Comment by MustacheMan2** [comment]
What an awesome object. Anyways, was able to save a model and it reappeared on reload. Thanks so much for looking into this rather than forgetting about it. Is this something that can occur for other users or is this just an odd case?

**Comment by EmosewaMC** [comment]
It was an edge case that was resolved by #911 so that migration wont fail anymore going forward
